src/components/FormioForm.spec.tsx

```diff
@@ -440,3 +441,116 @@ describe('Updating form errors', () => {
     });
   });
 });
+
+describe('Regressions', () => {
+  const NESTED_EDITGRIDS: EditGridComponentSchema = {
+    id: 'outer',
+    type: 'editgrid',
+    key: 'outer',
+    label: 'Outer',
+    groupLabel: 'Outer item',
+    disableAddingRemovingRows: true,
+    components: [
+      {
+        id: 'inner',
+        type: 'editgrid',
+        key: 'inner',
+        label: 'inner',
+        groupLabel: 'Inner item',
+        disableAddingRemovingRows: true,
+        components: [
+          {
+            type: 'textfield',
+            key: 'trigger',
+            id: 'trigger',
+            label: 'Checkbox',
+          },
+          {
+            type: 'textfield',
+            id: 'content2',
+            key: 'content2',
+            label: 'Not displayed 2',
+            conditional: {
+              show: false,
+              when: 'outer.inner.trigger',
+              eq: 'hide',
+            },
+          },
+        ],
+      },
+    ],
+  };
+
+  test('Prevent infinite render loop with nested editgrids', async () => {
+    const consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(err => {
+      throw new Error(err);
+    });
+    render(
+      <Form
+        components={[NESTED_EDITGRIDS]}
+        values={{
+          outer: [
+            {
+              inner: [
+                {
+                  trigger: 'hide',
+                  content2: 'Hide me',
+                },
+              ],
+            },
+          ],
+        }}
+        onSubmit={vi.fn()}
+      />
+    );
+
+    const editButton = await screen.findByRole('button', {name: 'Edit item 1'});
+
+    let error: Error | undefined = undefined;
+    try {
+      await userEvent.click(editButton);
+    } catch (e) {
+      error = e;
+      // if an error happens (which shouldn't be the case), make sure we're checking
+      // for the right kind of error!
+      expect(error!.message).toMatch(/Maximum update depth exceeded/);
+    }
+
+    expect(consoleErrorSpy.mock.calls).toHaveLength(0);
+    expect(error).toBeUndefined();
+
+    consoleErrorSpy.mockRestore();
+  });
+
+  test('Properly clears hidden fields in nested editgrids', async () => {
```

Comment:
@robinmolen you were [right](https://github.com/open-formulieren/formio-renderer/pull/120#discussion_r2221894951) :sweat_smile: